integration/fuse/src/main/java/alluxio/fuse/AlluxioFuseFileSystem.java

```diff
@@ -238,16 +238,14 @@ public int create(String path, @mode_t long mode, FuseFileInfo fi) {
     LOG.trace("create({}, {}) [Alluxio: {}]", path, Integer.toHexString(flags), uri);
 
     try {
+      if (mOpenFiles.size() >= MAX_OPEN_FILES) {
+        LOG.error("Cannot open {}: too many open files (MAX_OPEN_FILES: {})", uri,
+            MAX_OPEN_FILES);
+        return -ErrorCodes.EMFILE();
+      }
```

Comment:
If we move this out of the synchronized block, what prevents the number of open files exceeding max after we check?

Reply:
Could we make the MAX_FILE be a soft limit? Will a little amount of exceed max cause a problem?

Reply:
Yeah I think a soft limit would work